Crash in indexing.query for large resultsets

Stacktrace attached.

```
09:22:39.943 [nioEventLoopGroup-6-1] (930b0b71c0dbbc801cbc2e79b4077384) I cloud.calls.server.RpcServer - Incoming call: CallDescription(indexing.query) (QueryRequest(query=FileQuery(roots=[<REMOVED>], fileNameQuery=[], fileNameExact=null, extensions=AllOf(allOf=[AnyOf(anyOf=[txt, docx, rtf, csv, pdf], negate=false)]), fileTypes=AllOf(allOf=[AnyOf(anyOf=[FILE, DIRECTORY], negate=false)]), fileDepth=null, size=null), sortBy=null, itemsPerPage=100, page=116))

09:22:39.958 [nioEventLoopGroup-6-1] (930b0b71c0dbbc801cbc2e79b4077384) I cloud.calls.server.RpcServer - Uncaught exception in handler for CallDescription(indexing.query)

09:22:39.959 [nioEventLoopGroup-6-1] (930b0b71c0dbbc801cbc2e79b4077384) I cloud.calls.server.RpcServer - ElasticsearchStatusException[Elasticsearch exception [type=search_phase_execution_exception, reason=all shards failed]]; nested: ElasticsearchException[Elasticsearch exception [type=illegal_argument_exception, reason=Result window is too large, from + size must be less than or equal to: [10000] but was [11700]. See the scroll api for a more efficient way to request large data sets. This limit can be set by changing the [index.max_result_window] index level setting.]]; nested: ElasticsearchException[Elasticsearch exception [type=illegal_argument_exception, reason=Result window is too large, from + size must be less than or equal to: [10000] but was [11700]. See the scroll api for a more efficient way to request large data sets. This limit can be set by changing the [index.max_result_window] index level setting.]];

at org.elasticsearch.rest.BytesRestResponse.errorFromXContent(BytesRestResponse.java:177)

at org.elasticsearch.client.RestHighLevelClient.parseEntity(RestHighLevelClient.java:1793)

at org.elasticsearch.client.RestHighLevelClient.parseResponseException(RestHighLevelClient.java:1770)

at org.elasticsearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:1527)

at org.elasticsearch.client.RestHighLevelClient.performRequest(RestHighLevelClient.java:1484)

at org.elasticsearch.client.RestHighLevelClient.performRequestAndParseEntity(RestHighLevelClient.java:1454)

at org.elasticsearch.client.RestHighLevelClient.search(RestHighLevelClient.java:970)

at dk.sdu.cloud.indexing.util.ElasticUtilKt.search(ElasticUtil.kt:79)

at dk.sdu.cloud.indexing.services.ElasticQueryService.query(ElasticQueryService.kt:344)

at dk.sdu.cloud.indexing.http.QueryController$configure$$inlined$with$lambda$1.invokeSuspend(QueryController.kt:20)

at dk.sdu.cloud.indexing.http.QueryController$configure$$inlined$with$lambda$1.invoke(QueryController.kt)

at dk.sdu.cloud.calls.server.RpcServer.handleIncomingCall(RpcServer.kt:373)

at dk.sdu.cloud.calls.server.IngoingHttpInterceptor$addCallListenerForCall$1$1$1$1.invokeSuspend(IngoingHttpInterceptor.kt:63)

at dk.sdu.cloud.calls.server.IngoingHttpInterceptor$addCallListenerForCall$1$1$1$1.invoke(IngoingHttpInterceptor.kt)

at io.ktor.util.pipeline.SuspendFunctionGun.loop(PipelineContext.kt:268)

at io.ktor.util.pipeline.SuspendFunctionGun.access$loop(PipelineContext.kt:67)

at io.ktor.util.pipeline.SuspendFunctionGun.proceed(PipelineContext.kt:141)

at io.ktor.util.pipeline.SuspendFunctionGun.execute(PipelineContext.kt:161)

at io.ktor.util.pipeline.Pipeline.execute(Pipeline.kt:27)

at io.ktor.routing.Routing.executeResult(Routing.kt:147)

at io.ktor.routing.Routing.interceptor(Routing.kt:34)

at io.ktor.routing.Routing$Feature$install$1.invokeSuspend(Routing.kt:99)

at io.ktor.routing.Routing$Feature$install$1.invoke(Routing.kt)

at io.ktor.util.pipeline.SuspendFunctionGun.loop(PipelineContext.kt:268)

at io.ktor.util.pipeline.SuspendFunctionGun.access$loop(PipelineContext.kt:67)

at io.ktor.util.pipeline.SuspendFunctionGun.proceed(PipelineContext.kt:141)

at io.ktor.features.CallLogging$Feature$install$1$invokeSuspend$$inlined$withMDC$1.invokeSuspend(CallLogging.kt:226)

at io.ktor.features.CallLogging$Feature$install$1$invokeSuspend$$inlined$withMDC$1.invoke(CallLogging.kt)

at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:91)

at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:156)

at kotlinx.coroutines.BuildersKt.withContext(Unknown Source)

at io.ktor.features.CallLogging$Feature$install$1.invokeSuspend(CallLogging.kt:230)

at io.ktor.features.CallLogging$Feature$install$1.invoke(CallLogging.kt)

at io.ktor.util.pipeline.SuspendFunctionGun.loop(PipelineContext.kt:268)

at io.ktor.util.pipeline.SuspendFunctionGun.access$loop(PipelineContext.kt:67)

at io.ktor.util.pipeline.SuspendFunctionGun.proceed(PipelineContext.kt:141)

at io.ktor.util.pipeline.SuspendFunctionGun.execute(PipelineContext.kt:161)

at io.ktor.util.pipeline.Pipeline.execute(Pipeline.kt:27)

at io.ktor.server.engine.DefaultEnginePipelineKt$defaultEnginePipeline$2.invokeSuspend(DefaultEnginePipeline.kt:118)

at io.ktor.server.engine.DefaultEnginePipelineKt$defaultEnginePipeline$2.invoke(DefaultEnginePipeline.kt)

at io.ktor.util.pipeline.SuspendFunctionGun.loop(PipelineContext.kt:268)

at io.ktor.util.pipeline.SuspendFunctionGun.access$loop(PipelineContext.kt:67)

at io.ktor.util.pipeline.SuspendFunctionGun.proceed(PipelineContext.kt:141)

at io.ktor.util.pipeline.SuspendFunctionGun.execute(PipelineContext.kt:161)

at io.ktor.util.pipeline.Pipeline.execute(Pipeline.kt:27)

at io.ktor.server.netty.NettyApplicationCallHandler$handleRequest$1.invokeSuspend(NettyApplicationCallHandler.kt:36)

at io.ktor.server.netty.NettyApplicationCallHandler$handleRequest$1.invoke(NettyApplicationCallHandler.kt)

at kotlinx.coroutines.intrinsics.UndispatchedKt.startCoroutineUndispatched(Undispatched.kt:55)

at kotlinx.coroutines.CoroutineStart.invoke(CoroutineStart.kt:111)

at kotlinx.coroutines.AbstractCoroutine.start(AbstractCoroutine.kt:154)

at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch(Builders.common.kt:54)

at kotlinx.coroutines.BuildersKt.launch(Unknown Source)

at io.ktor.server.netty.NettyApplicationCallHandler.handleRequest(NettyApplicationCallHandler.kt:26)

at io.ktor.server.netty.NettyApplicationCallHandler.channelRead(NettyApplicationCallHandler.kt:20)

at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374)

at io.netty.channel.AbstractChannelHandlerContext.access$600(AbstractChannelHandlerContext.java:56)

at io.netty.channel.AbstractChannelHandlerContext$7.run(AbstractChannelHandlerContext.java:365)

at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)

at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:416)

at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:515)

at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)

at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)

at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)

at java.base/java.lang.Thread.run(Unknown Source)

Suppressed: org.elasticsearch.client.ResponseException: method [POST], host [http://elasticsearch:9200], URI [/files/_search?typed_keys=true&ignore_unavailable=false&expand_wildcards=open&allow_no_indices=true&ignore_throttled=true&search_type=query_then_fetch&batched_reduce_size=512&ccs_minimize_roundtrips=true], status line [HTTP/1.1 400 Bad Request]

{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"Result window is too large, from + size must be less than or equal to: [10000] but was [11700]. See the scroll api for a more efficient way to request large data sets. This limit can be set by changing the [index.max_result_window] index level setting."}],"type":"search_phase_execution_exception","reason":"all shards failed","phase":"query","grouped":true,"failed_shards":[{"shard":0,"index":"files","node":"zcwdy68CRha45pE6yWih5Q","reason":{"type":"illegal_argument_exception","reason":"Result window is too large, from + size must be less than or equal to: [10000] but was [11700]. See the scroll api for a more efficient way to request large data sets. This limit can be set by changing the [index.max_result_window] index level setting."}}],"caused_by":{"type":"illegal_argument_exception","reason":"Result window is too large, from + size must be less than or equal to: [10000] but was [11700]. See the scroll api for a more efficient way to request large data sets. This limit can be set by changing the [index.max_result_window] index level setting.","caused_by":{"type":"illegal_argument_exception","reason":"Result window is too large, from + size must be less than or equal to: [10000] but was [11700]. See the scroll api for a more efficient way to request large data sets. This limit can be set by changing the [index.max_result_window] index level setting."}}},"status":400}

at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:283)

at org.elasticsearch.client.RestClient.performRequest(RestClient.java:261)

at org.elasticsearch.client.RestClient.performRequest(RestClient.java:235)

at org.elasticsearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:1514)

... 60 more

Caused by: ElasticsearchException[Elasticsearch exception [type=illegal_argument_exception, reason=Result window is too large, from + size must be less than or equal to: [10000] but was [11700]. See the scroll api for a more efficient way to request large data sets. This limit can be set by changing the [index.max_result_window] index level setting.]]; nested: ElasticsearchException[Elasticsearch exception [type=illegal_argument_exception, reason=Result window is too large, from + size must be less than or equal to: [10000] but was [11700]. See the scroll api for a more efficient way to request large data sets. This limit can be set by changing the [index.max_result_window] index level setting.]];

at org.elasticsearch.ElasticsearchException.innerFromXContent(ElasticsearchException.java:496)

at org.elasticsearch.ElasticsearchException.fromXContent(ElasticsearchException.java:407)

at org.elasticsearch.ElasticsearchException.innerFromXContent(ElasticsearchException.java:437)

at org.elasticsearch.ElasticsearchException.failureFromXContent(ElasticsearchException.java:603)

at org.elasticsearch.rest.BytesRestResponse.errorFromXContent(BytesRestResponse.java:169)

... 63 more

Caused by: ElasticsearchException[Elasticsearch exception [type=illegal_argument_exception, reason=Result window is too large, from + size must be less than or equal to: [10000] but was [11700]. See the scroll api for a more efficient way to request large data sets. This limit can be set by changing the [index.max_result_window] index level setting.]]

at org.elasticsearch.ElasticsearchException.innerFromXContent(ElasticsearchException.java:496)

at org.elasticsearch.ElasticsearchException.fromXContent(ElasticsearchException.java:407)

at org.elasticsearch.ElasticsearchException.innerFromXContent(ElasticsearchException.java:437)

... 67 more
```

--- a/backend/indexing-service/build.gradle.kts
+++ b/backend/indexing-service/build.gradle.kts
@@ -1,4 +1,4 @@
-version = "1.16.4"
+version = "1.16.5-3"
 
 application {
     mainClassName = "dk.sdu.cloud.indexing.MainKt"

--- a/backend/indexing-service/k8.kts
+++ b/backend/indexing-service/k8.kts
@@ -3,7 +3,7 @@ package dk.sdu.cloud.k8
 
 bundle { ctx ->
     name = "indexing"
-    version = "1.16.4"
+    version = "1.16.5-3"
 
     withAmbassador {}
 

--- a/backend/indexing-service/src/main/kotlin/indexing/services/ElasticQueryService.kt
+++ b/backend/indexing-service/src/main/kotlin/indexing/services/ElasticQueryService.kt
@@ -1,6 +1,7 @@
 package dk.sdu.cloud.indexing.services
 
 import com.fasterxml.jackson.module.kotlin.readValue
+import dk.sdu.cloud.calls.RPCException
 import dk.sdu.cloud.defaultMapper
 import dk.sdu.cloud.file.api.StorageFile
 import dk.sdu.cloud.file.api.joinPath
@@ -20,6 +21,7 @@ import dk.sdu.cloud.indexing.util.search
 import dk.sdu.cloud.service.Loggable
 import dk.sdu.cloud.service.NormalizedPaginationRequest
 import dk.sdu.cloud.service.Page
+import io.ktor.http.HttpStatusCode
 import mbuhot.eskotlin.query.compound.bool
 import mbuhot.eskotlin.query.fulltext.match_phrase_prefix
 import mbuhot.eskotlin.query.term.range
@@ -52,6 +54,9 @@ class ElasticQueryService(
         paging: NormalizedPaginationRequest? = null,
         sorting: SortRequest? = null
     ): Page<ElasticIndexedFile> {
+        if (paging != null && (paging.page * paging.itemsPerPage + paging.itemsPerPage) > 10000) {
+            throw RPCException.fromStatusCode(HttpStatusCode.NotFound)
+        }
         return elasticClient.search<ElasticIndexedFile>(mapper, paging, FILES_INDEX) {
             if (sorting != null) {
                 val field = when (sorting.field) {
@@ -69,11 +74,18 @@ class ElasticQueryService(
 
             }
             if (paging != null) {
-                from(paging.page * paging.itemsPerPage)
-                size(paging.itemsPerPage)
+                if (paging.page * paging.itemsPerPage + paging.itemsPerPage <= 10000) {
+                    from(paging.page * paging.itemsPerPage)
+                    size(paging.itemsPerPage)
+                } else {
+                    size(10000)
+                }
+            } else {
+                //If pagination is not given then limit to 10000 results. If more needed use scroll API.
+                size(10000)
             }
             searchBasedOnQuery(query).also {
-                log.debug(it.toString())
+                log.info(it.toString())
             }
         }
     }

